### ISSUE TITLE
`read_csv_auto` converts column to double despite given column type as decimal

### ISSUE THREAD

**Issue by darthf1** [duckdb/duckdb#8573]
### What happens?

When I read a csv file with `read_csv_auto`, which has a column with a big number, and I define the column type as `DECIMAL(25,3)` it still gets converted to float, before it gets converted to DECIMAL which leads to precision loss.

I get the expected result with just `read_csv`, or by defining `ALL_VARCHAR=TRUE` with `read_csv_auto` . however I would like to keep using `read_csv_auto` without that parameter so other columns which have datetime varchars can have their format correctly detected by duckdb.

### To Reproduce

A direct SELECT statement via CLI gives the expected result:
```
v0.8.1 6536a77232
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.

D SELECT '649686000000001631.440'::DECIMAL(25,3) AS 'bignumber';
┌────────────────────────┐
│       bignumber        │
│     decimal(25,3)      │
├────────────────────────┤
│ 649686000000001631.440 │
└────────────────────────┘
```

When I put this in a CSV, and load it through CLI with `read_csv`, I still get the expected result:
```
bignumber
82544000000000707.52
```

```
D SELECT bignumber, "bignumber"::DECIMAL(25,3) AS "bignumber" FROM read_csv('test.csv', HEADER=TRUE, COLUMNS={'bignumber': 'DECIMAL(25,3)'}, QUOTE='"', DELIM=',');
┌────────────────────────┬────────────────────────┐
│       bignumber        │       bignumber        │
│     decimal(25,3)      │     decimal(25,3)      │
├────────────────────────┼────────────────────────┤
│ 649686000000001631.440 │ 649686000000001631.440 │
└────────────────────────┴────────────────────────┘
```

When I use `read_csv_auto` it fails: 
```
D SELECT bignumber, "bignumber"::DECIMAL(25,3) AS "bignumber" FROM read_csv_auto('test.csv', HEADER=TRUE, COLUMNS={'bignumber': 'DECIMAL(25,3)'}, QUOTE='"', DELIM=',');
┌───────────────────────┬────────────────────────┐
│       bignumber       │       bignumber        │
│        double         │     decimal(25,3)      │
├───────────────────────┼────────────────────────┤
│ 6.496860000000017e+17 │ 649686000000001638.400 │
└───────────────────────┴────────────────────────┘
```

### OS:

Ubuntu 22.04.2 LTS

### DuckDB Version:

v0.8.1 and 0.8.2.dev2809

### DuckDB Client:

Python

### Full Name:
Frank Verschuren

### Affiliation:
1 Giant Leap Solutions BV

### Have you tried this on the latest `master` branch?

I have tested with a master build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by Tishj** [comment]
It would be great if you could test this on #8253, this sounds like it might be fixed by it.

**Comment by darthf1** [comment]
I tried, it gives the same result:

```
git clone https://github.com/pdet/duckdb.git
cd duckdb
git checkout csv_sniffer
make
./build/release/duckdb
```

```
v0.2.4-dev25592 d6733ec07a
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.

D SELECT bignumber, "bignumber"::DECIMAL(25,3) AS "bignumber" FROM read_csv_auto('test.csv', HEADER=TRUE, COLUMNS={'bignumber': 'DECIMAL(25,3)'}, QUOTE='"', DELIM=',');
Error: IO Error: No files found that match the pattern "test.csv"
D SELECT bignumber, "bignumber"::DECIMAL(25,3) AS "bignumber" FROM read_csv_auto('/home/frank/downloads/test.csv', HEADER=TRUE, COLUMNS={'bignumber': 'DECIMAL(25,3)'}, QUOTE='"', DELIM=',');
┌───────────────────────┬────────────────────────┐
│       bignumber       │       bignumber        │
│        double         │     decimal(25,3)      │
├───────────────────────┼────────────────────────┤
│ 6.496860000000017e+17 │ 649686000000001638.400 │
└───────────────────────┴────────────────────────┘

**Comment by darthf1** [comment]
It does work if I override the columns type via the `TYPE` parameter:
```
D SELECT bignumber, "bignumber"::DECIMAL(25,3) AS "bignumber" FROM read_csv_auto('test.csv', HEADER=TRUE, COLUMNS={'bignumber': 'DECIMAL(25,3)'}, TYPES={'bignumber': 'DECIMAL(25,3)'}, QUOTE='"', DELIM=',');
┌────────────────────────┬────────────────────────┐
│       bignumber        │       bignumber        │
│     decimal(25,3)      │     decimal(25,3)      │
├────────────────────────┼────────────────────────┤
│ 649686000000001631.440 │ 649686000000001631.440 │
└────────────────────────┴────────────────────────┘
```

So maybe it does work as intended? It looks like using `TYPES` is the same as using the `read_csv` instead of `read_csv_auto`, where you can override the type for specific columns.

**Comment by Tishj** [comment]
```c++
	if (options.auto_detect) {
		options.file_path = result->files[0];
		auto initial_reader = make_uniq<BufferedCSVReader>(context, options);
		return_types.assign(initial_reader->return_types.begin(), initial_reader->return_types.end());
		if (names.empty()) {
			names.assign(initial_reader->names.begin(), initial_reader->names.end());
		} else {
			if (explicitly_set_columns) {
				// The user has influenced the names, can't assume they are valid anymore
				if (return_types.size() != names.size()) {
					throw BinderException("The amount of names specified (%d) and the observed amount of types (%d) in "
					                      "the file don't match",
					                      names.size(), return_types.size());
				}
			} else {
				D_ASSERT(return_types.size() == names.size());
			}
			initial_reader->names = names;
		}
		options = initial_reader->options;
		result->initial_reader = std::move(initial_reader);
```

It seems we just disregard the explicitly set types through "columns":
`return_types.assign(initial_reader->return_types.begin(), initial_reader->return_types.end());`

`return_types` contained the DECIMAL type before this

**Comment by hannes** [comment]
Hey @darthf1 you seem to have edited the issue template to remove the fields for name and affiliation. Would you mind stating those please? Thanks!

**Comment by darthf1** [comment]
@hannes Yes, done

**Comment by darthf1** [comment]
Resolved in https://github.com/duckdb/duckdb/pull/8852 and available in `v0.9.0`
